### ISSUE TITLE
Manifest application error for kubernetes 1.17.3

### ISSUE THREAD

**Issue by lukmy** [spinnaker/spinnaker#5715]
### Issue Summary:

Deploy (Manifest) failure for kubernetes 1.17.3

### Cloud Provider(s):

- kubernetes v2
- kubernetes version: 1.17.3

### Environment:

- spinnaker: 1.19.2

### Feature Area:

- clouddriver
- kubernetes v2

### Description:

Pipeline Execution Error:

![image](https://user-images.githubusercontent.com/1225713/80476388-e8a07000-897c-11ea-99db-9fdb581071e6.png)

My configuration:

![image](https://user-images.githubusercontent.com/1225713/80476318-ceff2880-897c-11ea-9e90-47af04161812.png)

We found that version of kubectl embedded in clouddriver is 1.16.0, while the error fields (matchPolicy, timeoutSeconds) are unavailable for kubectl with version 1.16.0, and the manifest validation blocked the pipeline. I think the validation is necessary, so the version of kubectl might be a problem. 

Are there any risks if I replace kubectl in clouddriver with version 1.17.3?

**Comment by ezimanyi** [comment]
@lukmy : Thanks for opening this issue.  For this specific issue I've bumped the version of `kubectl` that we ship with `clouddriver` to 1.17.x in the above PR (which will be in Spinnaker 1.21).

To your general question, I think that in general there should not be any issue for you to manually replace the version in the container.  We're not in general using a lot of alpha features that are likely to depend precisely on the version (and I'm not aware of us ever having an issue when we upgraded the version we ship with the container).

The main reason we don't usually upgrade right away is because Kubernetes offers +/-1 minor version skew between `kubectl` and the cluster, and we'd like to maximize that overlap with what users are running.  (Which is why I bumped to 1.17 above even though 1.18 is out.). That being said  I do think the actual usable skew between `kubectl` and the cluster is in general much larger than the officially supported +/- 1 version.

So in short, if the version that ships with the container doesn't work for your particular cluster, I don't think you'll run into any issues installing a more suitable version.

**Comment by lukmy** [comment]
> @lukmy : Thanks for opening this issue. For this specific issue I've bumped the version of `kubectl` that we ship with `clouddriver` to 1.17.x in the above PR (which will be in Spinnaker 1.21).
> 
> To your general question, I think that in general there should not be any issue for you to manually replace the version in the container. We're not in general using a lot of alpha features that are likely to depend precisely on the version (and I'm not aware of us ever having an issue when we upgraded the version we ship with the container).
> 
> The main reason we don't usually upgrade right away is because Kubernetes offers +/-1 minor version skew between `kubectl` and the cluster, and we'd like to maximize that overlap with what users are running. (Which is why I bumped to 1.17 above even though 1.18 is out.). That being said I do think the actual usable skew between `kubectl` and the cluster is in general much larger than the officially supported +/- 1 version.
> 
> So in short, if the version that ships with the container doesn't work for your particular cluster, I don't think you'll run into any issues installing a more suitable version.

Thank you for replying the issue. 

It seems that the error is not raised by kubernetes cluster, but kubectl. If we use --validate=false, it could be applied.
